### ISSUE TITLE
TWRP-3.4.0-0-parker Backup is not restored!

### ISSUE THREAD

**Issue by ilia3367** [TeamWin/android_bootable_recovery#61]
- [*] I am running an official build of TWRP, downloaded from https://twrp.me/motorola/motorolaonezoom.html
- [*] I am running the latest version of TWRP
- [*] I have read the FAQ (https://twrp.me/FAQ/)
- [*] I have searched for my issue and it does not already exist

**Device codename**: Motorola One Zoom. Codename - parker.
**TWRP version**: 3.4.0-0

New TWRP twrp-3.4.0-0-parker Backup is created but not restored. The system won't start. Cyclical reboot with red warning.
Checked with the firmware XT2010-1_PARKER_RETEU_9.0_PPHS29.59-51-13.
I'm asking for help in solving this problem!

**Comment by betaxab** [comment]
You should choose "System Image" method to backup your system partition.
Or try to remove DM-Verity & motorola hab/verify protect, then you can use "System" method, OK? It's not TWRP's bug.

![Screenshot_2020-08-22-16-19-56](https://user-images.githubusercontent.com/30405429/90952097-5bd17300-e493-11ea-9585-d8ba3c556e26.png)

**Comment by ilia3367** [comment]
Hi!
I tried make backup and restoring System partition and System.img, Vendor partition and
Vendor.img. The result is the same - the system won't start.
TWRP from mauronofrio works well.

сб, 22 авг. 2020 г. в 11:38, betaxab <notifications@github.com>:

**Comment by betaxab** [comment]
Please try to remove DM-Verity & motorola hab/verify protect, then you can use "System" method.

**Comment by ilia3367** [comment]
> Please try to remove DM-Verity & motorola hab/verify protect, then you can use "System" method.

I use Disable_Dm-Verity_ForceEncrypt https://forum.xda-developers.com/android/software/universal-dm-verity-forceencrypt-t3817389

**Comment by ilia3367** [comment]
This TWRP is working well https://forum.xda-developers.com/motorola-one-pro/development/twrp-3-3-1-root-motorola-one-zoom-t4089931

**Comment by betaxab** [comment]
No! try to remove motorola hab/verify protect. There have a blog, try to use modified magisk: https://www.htcp.net/4924.html

**Comment by ilia3367** [comment]
This is some kind of very secret Magisk :)
OK! I'll try and write you about the results.
But I don't think it's going to help...

**Comment by ilia3367** [comment]
In my opinion, TWRP should back up the system and restore it even without Root and Magisk.

**Comment by betaxab** [comment]
Well, it is not associated with Magisk. Motorola hab verify will prevent boot with corrupted System image signature.

There is a similar file here: https://github.com/boulzordev/motorola_deen_sprout_dump/blob/user-9-PPKS29.68-16-33-5-e8ba0-release-keys/vendor/etc/init/hw/init.mmi.hab.rc

**Comment by ilia3367** [comment]
I did as you advised and the system still won't boot, red warning.
I enclose screenshots and TWRP logs. 
https://drive.google.com/file/d/1AdRZNNjMKoL2iJFrZu_ehvpV1GcmvWw0/view?usp=sharing

**Comment by ilia3367** [comment]
My course of action:
1. I'm reinstalling the stock firmware.
2. Boot the phone in fastboot mode, connect it to a PC and perform the following commands:
fastboot boot twrp-3.4.0-0-parker.img
3. Install ttwrp-installer-3.4.0-0-parker.zip 
5. Format data for decryption.
6. Reboot Recovery.
7. Install Magisk-disable-dm-verity-v20.4.zip  https://www.lanzous.com/iaurp5c 
8. Loading into the system, making all the settings.
9. Reboot to TWRP, make a backup and restore from it.

So, I don't think DM-Verity and Magisk are the problem here.

**Comment by betaxab** [comment]
Your log does not include flash Magisk-disable-dm-verity-v20.4.zip part.

**Comment by ilia3367** [comment]
> Your log does not include flash Magisk-disable-dm-verity-v20.4.zip part.

There will be no information about Magisk installation in this log. The log was copied after restarting the system when I was backing up and restoring.

You don't believe me? I installed it and took a screenshot. Take a look at my archive.
There's no point in me lying to you.
You made this TWRP official. I tested it. Found some mistakes. But you don't want to admit that the problem is TWRP itself.
TWRP must restore the backup without Magisk.

**Comment by ilia3367** [comment]
Excuse me! Accidentally clicked the Close Issue button

**Comment by betaxab** [comment]
Please try:
1. Flash full firmware zip.
2. Boot to fastboot mode, run fastboot boot twrp-3.4.0-0-parker.img
3. Flash Magisk-disable-dm-verity-v20.4.zip
4. Backup system & vendor & boot
5. Restore system & vendor & boot

Don't touch & Backup/Restore Oem partition.

**Comment by ilia3367** [comment]
> Don't touch & Backup/Restore Oem partition.

Mauronofrio also had a problem with the oem section. But he corrected it quickly. Could you ask him how to fix this mistake?
What about the other partitions? I mean, we need a full backup.

**Comment by betaxab** [comment]
Could you understand this line?
https://github.com/mauronofrio/android_device_motorola_parker/blob/android-9.0/recovery/root/etc/twrp.fstab#L17

"emmc" means `Image method` backup & restore instand of regular `Tar method`.
`Image method` will backup a full partition to an image file (like `dd if=/dev/block/bootdevice/by-name/oem of=/sdcard/oem.img`), of course it will include the Motorola image signature, so it will not has verification issues.
I don't think so it is TWRP's bug, it is actually an issue that Motorola verity protect is not completely disabled. Maybe in the future custom rom/kernel developer will be found a way to solve Motorola verity protect for OEM partition.

I can add an `Image method` Backup/Restore oem partition for parker, please wait for TWRP to release an new version.

**Comment by ilia3367** [comment]
Yes!!! The reason is oem partition!!!
I didn't install Magisk. Backup and restore was successful!
This has nothing to do with Magisk.
Backing up and restoring System and Vendor partitions, not .img, does not work. 
Can I remove these partitions from the partition selection list?

**Comment by betaxab** [comment]
I said, it's motorola hab protect issue, I just modified Magisk Installer zip let remove Motorola hab protect easily.

I have added `Image method` for oem partition, you can download it from lolinet (or waiting for TWRP to release a new version):
https://mirrors.lolinet.com/firmware/twrp/parker/

Please use `Image method` to Backup & Restore Unmodified System/Vendor/Oem Partition if you didn't completely disabled Motorola Hab Verity Protect.

----

By the way, It support FBE Encryption not need to format data, encryption will not cause obviously performance loss.

**Comment by ilia3367** [comment]
Thank you for Magisk! Thank you for TWRP! Thank you for hearing me!
I will definitely make changes in the Motorola One Zoom 4PDA and XDA themes. 
I will wait for the corrected version of TWRP on the official website.
Give me a hint, please! Will need to new  build  TWRP for firmware  on Android 10, which should be released soon?
